src/Atmos/Model/boundaryconditions.jl

```diff
@@ -71,12 +76,17 @@ Set the momentum at the boundary to be zero.
 struct NoFluxBC <: BoundaryCondition
 end
 
-function atmos_boundary_state!(::Rusanov, bc::NoFluxBC, m::AtmosModel,
-                               stateP::Vars, auxP::Vars, nM, stateM::Vars,
-                               auxM::Vars, bctype, t, _...)
+function atmos_boundary_state!(nf::Union{Rusanov, CentralNumericalFluxGradient},
+                               bc::NoFluxBC, m::AtmosModel, stateP::Vars,
+                               auxP::Vars, nM, stateM::Vars, auxM::Vars, bctype,
+                               t, _...)
   FT = eltype(stateM)
   stateP.ρ = stateM.ρ
-  stateP.ρu -= 2 * dot(stateM.ρu, nM) * SVector(nM)
+  if typeof(nf) == Rusanov
+    stateP.ρu -= 2 * dot(stateM.ρu, nM) * SVector(nM)
```

Comment:
why is this necessary?

Reply:
This is because we run this non-diffussive flux back through the numerical flux function, wheras for the Gradient terms it is used directly in the flux function.

@fxgiraldo can you confirm this is right?

Reply:
Do we need this special handling for any other variables, or just momentum?

Reply:
Just momentum for this boundary condition type

Can you confirm @fxgiraldo